### PR TITLE
small fixes for python3

### PR DESCRIPTION
**Pull request by gpoulin**
Various fixes for python3. Based on my tests with luigi, those modifications are enough to make the minicluster work with python3.

### CODE REVIEW

File: snakebite/client.py [hunk 1]
```diff
@@ -40,6 +40,10 @@
 import errno
 import time
 import re
+import sys
+
+if sys.version > '3':
```
Comment (Tarrasch) [inline snakebite/client.py]: This looks both a bit scary (since you're not using a library like six) and wrong (you meant  `>=` I guess? :)).

**Comment by Tarrasch** [comment]
Oh, and if you decide to go for python 3, please install six like we did for luigi, as otherwise it won't build on debian. :)

**Comment by gpoulin** [comment]
@Tarrasch In fact, it's not wrong, it's even the documented way http://python3porting.com/noconv.html#integer-incompatibilities. I admit that it looks wrong thou.

I don't think it's necessary to import six here. I imported six in Luigi because Luigi has some complex structure like metaclass. In addition, the issue with debian was that the six version in debian was too old. By not importing six we don't have this issue. In addition, adding six here wouldn't be so useful since there isn't anything in six to support long vs int.

I'll replace `if sys.version > '3'` by `sys.version_info[0] == 3` which is clearer and it's what six use https://bitbucket.org/gutworth/six/src/986089f3d04e468458cc9f5b9b3be26fb3d75ade/six.py?at=default#cl-37.

The modification to `snakebite/client.py` and `snakebite/formatter.py` are purely syntactic. For `snakebite/minicluster.py`, the main difference is that it open the the process pipes in text mode (`universal_newlines=True`) for processes that expect text in stdin and output text in stdout. For processes that might output bytes, `cat` and `tail` (let me know if you think I missed one), the pipes are still open in bytes mode by default (for backward compatibility), but can can be open in text mode.

P.S.: The main goal here was just to support the minicluster to be able to support hdfs test on python3 for luigi.

**Comment by Tarrasch** [comment]
Ok, you seem to be well aware of what you're doing. I would have merged this but I'll wait for the actual maintainers of this project to do so. Pinging @ravwojdyla

**Comment by ravwojdyla** [comment]
Hi @gpoulin @Tarrasch 

Sorry - been off for some time.
Thanks for PR! Looks good to me!

While some of this may be out of the scope of purpose of this change, some minor comments:

1. mode:
 * could you update comment in client about value of mode [here](https://github.com/spotify/snakebite/blob/master/snakebite/client.py#L166)


1. division:

 * do you think it makes sense to update division operators in channel [here](https://github.com/spotify/snakebite/blob/master/snakebite/channel.py#L594) and [here](https://github.com/spotify/snakebite/blob/master/snakebite/channel.py#L595)
 * does [it](https://github.com/spotify/snakebite/blob/master/test/util.py#L38) require less ambiguous division operator?

**Comment by gpoulin** [comment]
@ravwojdyla I updated the docstring.

For the division in channel, it won't makes difference for the interpreter. On code readability, I'm not sure which is best. It took me some time to understand that the division was a real division and not a integer division. If you think readability is important, I would rather add a comment or change to something like `int(float(bytes_per_load) / bytes_per_chunk)`.

For the division ambiguity, `fromtimestamp` accept both int and float. In this situation, I don't think ±1s  make so much difference, we truncated at the hour level.

I need to admit that I didn't look so much out of the minicluster module. I just looked that python wasn't yelling when trying to import.

**Comment by ravwojdyla** [comment]
Ok - let's unblock this and merge it - #142 will focus on making sure the py3 style is applied for whole snakebite code.

**Comment by ravwojdyla** [comment]
Thanks @gpoulin

**Pull request merged**